File: polimorfismo/java/FreeLancer.java
// Disciplina: Programação Orientada a Objetos
// Professor : Gilson Pereira
// Assunto   : Polimorfismo em Java
// Fonte     : Ascencio e Campos - Fundamentos da Programação de Computadores (3ª ed.)

public class FreeLancer extends Funcionario {
	private int diasTrabalhados;
	private float valorDia;
	
	public FreeLancer() {
		
	}
	
	public int getDiasTrabalhados() {
		return diasTrabalhados;
	}
	
	public void setDiasTrabalhados(int diasTrabalhados) {
		this.diasTrabalhados = diasTrabalhados;
	}
	
	public float getValorDia() {
		return valorDia;
	}
	
	public void setValorDia(float valorDia) {
		this.valorDia = valorDia;
	}
	
	@Override
	public float calcularSalarioFinal() {
		return diasTrabalhados * valorDia;
	}
}

File: polimorfismo/cpp/FreeLancer.cpp
// Disciplina: Programação Orientada a Objetos
// Professor : Gilson Pereira
// Assunto   : Polimorfismo em C++
// Fonte     : Ascencio e Campos - Fundamentos da Programação de Computadores (3ª ed.)

#include "FreeLancer.hpp"
#include <iostream>

FreeLancer::FreeLancer()
{

}

int FreeLancer::getDiasTrabalhados()
{
    return diasTrabalhados;
}

void FreeLancer::setDiasTrabalhados(int d)
{
    diasTrabalhados = d;
}

float FreeLancer::getValorDia()
{
    return valorDia;
}

void FreeLancer::setValorDia(float v)
{
    valorDia = v;
}

float FreeLancer::calcularSalarioFinal()
{
    return diasTrabalhados * valorDia;
}